src/mscorlib/shared/System/ArgumentException.cs

```diff
@@ -64,13 +64,12 @@ public ArgumentException(String message, String paramName)
         protected ArgumentException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            _paramName = info.GetString("ParamName");
+            throw new PlatformNotSupportedException();
```

Comment:
We discussed this offline and decided to continue throwing from exception serialization constructors. I'll put together a PR to remove the three DCS tests that test ArgumentException.